processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferGrouper.java

```diff
@@ -428,10 +431,17 @@ private void growIfPossible()
     final ByteBuffer entryBuffer = tableBuffer.duplicate();
     final ByteBuffer keyBuffer = tableBuffer.duplicate();
 
+    List<Map<Integer, Integer>> reloationMaps = new ArrayList<>(aggregators.length);
+    for (int i = 0; i < aggregators.length ; i++)
+    {
+      reloationMaps.add(new HashMap<Integer, Integer>(buckets));
```

Comment:
ok, but then BufferAggregator will assume that no position collisions can happen.

but i see the point about keeping most memory usage to be off-heap.

@akashdw  let us do the non-batching version then.